src/OpenSpaceToolkit/Astrodynamics/Flight/System/Dynamics/SatelliteDynamics.cpp

```diff
@@ -0,0 +1,276 @@
+{
+
+    displayDecorator ? ostk::core::utils::Print::Header(anOutputStream, "SatelliteDynamics") : void () ;
+
+    ostk::core::utils::Print::Line(anOutputStream) << "Environment:" << environment_ ;
+
+    ostk::core::utils::Print::Separator(anOutputStream, "Satellite System") ;
+    satelliteSystem_.print(anOutputStream, false) ;
+
+    ostk::core::utils::Print::Separator(anOutputStream, "State") ;
+    state_.print(anOutputStream, false) ;
+
+    displayDecorator ? ostk::core::utils::Print::Footer(anOutputStream) : void () ;
+
+}
+
+State                           SatelliteDynamics::getState                 ( ) const
+{
+
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Satellite Dynamics") ;
+    }
+
+    return state_ ;
+
+}
+
+void                            SatelliteDynamics::setState                 (   const   State&                      aState                                      )
+{
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Satellite Dynamics") ;
+    }
+
+    state_ = aState ;
+    environment_.setInstant(aState.getInstant()) ;
+
+}
+
+Dynamics::DynamicalEquationWrapper  SatelliteDynamics::getDynamicalEquations( )
+{
+
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("SatelliteDynamics") ;
+    }
+
+    return std::bind(&SatelliteDynamics::DynamicalEquations, this, std::placeholders::_1,  std::placeholders::_2,  std::placeholders::_3) ;
+
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+void                            SatelliteDynamics::DynamicalEquations       (   const   Dynamics::StateVector&      x,
+                                                                                        Dynamics::StateVector&      dxdt,
+                                                                                const   double                      t                                           )
+{
+
+    const Position currentPosition = Position::Meters({ x[0], x[1], x[2] }, gcrfSPtr_) ;
+
+    // Update environment time
+    const Instant currentInstant = state_.getInstant() + Duration::Seconds(t) ;
+    environment_.setInstant(currentInstant) ;
+
+    // Initialize gravitational acceleration vector
+    Vector3d totalGravitationalAcceleration_SI = {0.0, 0.0, 0.0};
+
+    // Access all objects in the environment and loop through them
+    Array<String> objectNameArray = environment_.getObjectNames() ;
+
+    for (size_t i = 0; i < (size_t)objectNameArray.getSize(); i++)
+    {
+        if (objectNameArray[i] == "Earth")
+        {
+            // Obtain gravitational acceleration from current object
+            const Vector gravitationalAcceleration = (environment_.accessCelestialObjectWithName(objectNameArray[i]))->getGravitationalFieldAt(currentPosition) ;
```

Comment:
No need for extra `()` + use `objectName` directly:

```suggestion
            const Vector gravitationalAcceleration = environment_.accessCelestialObjectWithName(objectName)->getGravitationalFieldAt(currentPosition) ;
```

Reply:
fixed